astropy/utils/tests/test_decorators.py

```diff
@@ -57,16 +59,29 @@ class DummyClass:
         def __init__(self):
             self._foo = 42
             self._bar = 4242
+            self._message = '42'
+            self._alternative = [42]
+            self._pending = {42}
```

Comment:
I don't grok why "alternative" is a list and not string, or "pending" is not bool here. Will need a second pair of eyes who are more familiar with this decorator.

The rest of the patch LGTM. Thanks!

Reply:
These variables share names with the arguments that get used when they are deprecated, but their values and types are irrelevant.